server/src/main/java/org/uiautomation/ios/instruments/commandExecutor/CURLIAutomationCommandExecutor.java

```diff
@@ -112,7 +116,24 @@ private void getResponse(HttpServletRequest request, HttpServletResponse respons
       if (request.getInputStream() != null) {
         StringWriter writer = new StringWriter();
         IOUtils.copy(request.getInputStream(), writer, "UTF-8");
-        String json = writer.toString();
+        String json = null;
+        String sessionUUID = null;
+        String jsonConcatenated = "";
+        if (writer.toString().endsWith(PARTIAL_IDENTIFIER)){
+            sessionUUID = writer.toString().substring(0, UUID_LENGTH);
+            if (jsonMap.get(sessionUUID) != null){
+                jsonConcatenated = (String)jsonMap.get(sessionUUID);
+            }
+            jsonConcatenated =  jsonConcatenated.concat(writer.toString().substring(UUID_LENGTH, writer.toString().length() - PARTIAL_IDENTIFIER.length()));
```

Comment:
You are constructing the json parts as

Each json part except last part = <36 character session id><Some part of the original response><44 character partial id>
Last json part                            = <36 character session id><Some part of the original response><44 character final id>

let's say writer.toString().length() is equal to n

  len(PARTIAL_IDENTIFIER) = 44

so you are doing

  jsonConcatenated =  jsonConcatenated.concat(writer.toString().substring(36, n - 44));

which implies 

  jsonConcatenated = <part 1 of original response>b94b2b80-4330-11e4-916c-0800200c9a66<part 2 of original response>b94b2b80-4330-11e4-916c-0800200c9a66...<part n of original response>b94b2b80-4330-11e4-916c-0800200c9a66>

I believe what you want is

  jsonConcatenated = <part 1 of original response><part 2 of original response>...<part n of original response>b94b2b80-4330-11e4-916c-0800200c9a66>